python/lsst/ts/electrometer/controller.py

```diff
@@ -330,11 +358,31 @@ def write_fits_file(self, intensity, times, temperature, unit):
         hdr["CLMN1"] = ("Time", "Time in seconds")
         hdr["CLMN2"] = "Intensity"
         filename = f"{self.manual_start_time}_{self.manual_end_time}.fits"
-        hdu.writeto(f"{self.file_output_dir}/{filename}")
-        self.log.info(f"Electrometer Scan data file written: {filename}")
-        self.log.info(
-            f"Scan Summary of [Mean, median, std] is: "
-            f"[{np.mean(intensity):0.5e}, {np.median(intensity):0.5e}, {np.std(intensity):0.5e}]"
+        try:
+            hdu.writeto(f"{self.file_output_dir}/{filename}")
+            self.log.info(f"Electrometer Scan data file written: {filename}")
+            self.log.info(
+                f"Scan Summary of [Mean, median, std] is: "
+                f"[{np.mean(intensity):0.5e}, {np.median(intensity):0.5e}, {np.std(intensity):0.5e}]"
+            )
+        except Exception:
+            self.log.exception("Writing file to local disk failed.")
+        try:
+            file_upload = io.BytesIO()
+            hdu.writeto(file_upload)
+            file_upload.seek(0)
+            key_name = self.csc.bucket.make_key(
+                salname="Electrometer",
+                salindexname=self.csc.index,
+                generator="fits",
+                date=astropy.time.Time(self.manual_end_time, format="unix"),
+                suffix=".fits",
+            )
+            await self.csc.bucket.upload(fileobj=file_upload, key=key_name)
+        except Exception:
+            self.log.exception("Uploading file to s3 bucket failed")
```

Comment:
Please raise here.

Right now your code continues and you report the largeFileObject event.